open-authoring/error-interpreter/client-side/notFound.ts

```diff
@@ -1,12 +1,13 @@
 import OpenAuthoringError from '../../OpenAuthoringError'
-import { isForkValid } from '../../github/api'
 import OpenAuthoringErrorProps from '../../OpenAuthoringErrorProps'
-import { enterAuthFlow, refresh, justClose } from '../actions'
+import { enterAuthFlow, refresh } from '../actions'
+import { getForkName, getHeadBranch } from '../../utils/repository'
 
 export default async function interpretNotFoundError(
-  error: OpenAuthoringError
+  error: OpenAuthoringError,
+  github: any
 ) {
-  if (await isForkValid(null)) {
+  if (await github.getBranch(getForkName(), getHeadBranch())) {
```

Comment:
Hmmm, just spitballing. Maybe we're missing some kind of repository object?

```
if (await repository.getBranch())
```

And the repository knows about the github API 🤔